File: php/features.php
<?php
declare(strict_types=1);

// KolnerAdressen SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KolnerAdressenFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KolnerAdressenBaseFeature();
            case "test":
                return new KolnerAdressenTestFeature();
            default:
                return new KolnerAdressenBaseFeature();
        }
    }
}

File: py/kolneradressen_sdk/features.py
# KolnerAdressen SDK feature factory

from kolneradressen_sdk.feature.base_feature import KolnerAdressenBaseFeature
from kolneradressen_sdk.feature.test_feature import KolnerAdressenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KolnerAdressenBaseFeature(),
        "test": lambda: KolnerAdressenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
